File: 1935-maximum-number-of-words-you-can-type/1935-maximum-number-of-words-you-can-type.js
/**
 * @param {string} text
 * @param {string} brokenLetters
 * @return {number}
 */
var canBeTypedWords = function(text, brokenLetters) {
    let words = text.split(" ")
    let num_of_words = words.length
    
    for (let word of words){
        for (let char of word){
            if (brokenLetters.includes(char)){
                num_of_words -= 1
                break
            }
        }
    }
    
    return num_of_words
};

File: 1935-maximum-number-of-words-you-can-type/1935-maximum-number-of-words-you-can-type.py
class Solution:
    def canBeTypedWords(self, text: str, brokenLetters: str) -> int:
        
        # Approach 1: O(words * brokenLetters)
#         words = text.split()
#         true_count = 0
#         for word in words:
#             if all(letter not in word for letter in brokenLetters):
#                 true_count +=1
            
#         return true_count
    
        # Approach 2
        words = text.split()
        num_of_words = len(words)
        
        for word in words:
            for char in word:
                if char in brokenLetters:
                    num_of_words -= 1
                    break
                    
        return num_of_words
        